modules/apps/object/object-service/src/main/java/com/liferay/object/internal/upgrade/v8_8_2/ObjectFolderGuestPermissionUpgradeProcess.java

```diff
@@ -0,0 +1,61 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.object.internal.upgrade.v8_8_2;
+
+import com.liferay.object.model.ObjectFolder;
+import com.liferay.object.service.ObjectFolderLocalService;
+import com.liferay.portal.kernel.model.ResourceConstants;
+import com.liferay.portal.kernel.model.Role;
+import com.liferay.portal.kernel.model.role.RoleConstants;
+import com.liferay.portal.kernel.security.permission.ActionKeys;
+import com.liferay.portal.kernel.service.CompanyLocalService;
+import com.liferay.portal.kernel.service.ResourcePermissionLocalService;
+import com.liferay.portal.kernel.service.RoleLocalService;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+
+/**
+ * @author Guilherme Sa
+ */
+public class ObjectFolderGuestPermissionUpgradeProcess extends UpgradeProcess {
+
+	public ObjectFolderGuestPermissionUpgradeProcess(
+		CompanyLocalService companyLocalService,
+		ObjectFolderLocalService objectFolderLocalService,
+		ResourcePermissionLocalService resourcePermissionLocalService,
+		RoleLocalService roleLocalService) {
+
+		_companyLocalService = companyLocalService;
+		_objectFolderLocalService = objectFolderLocalService;
+		_resourcePermissionLocalService = resourcePermissionLocalService;
+		_roleLocalService = roleLocalService;
+	}
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		_companyLocalService.forEachCompany(
+			company -> {
+				Role guestRole = _roleLocalService.getRole(
+					company.getCompanyId(), RoleConstants.GUEST);
+
+				ObjectFolder objectFolder =
```

Comment:
The `objectFolder` variable should be declared before the `guestRole` because in this case we prioritize the order of use. [See an example](https://github.com/liferay/liferay-portal/blob/6ce0b9eb049e99a50285a324a3bd4afd76d340ab/modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/upgrade/v3_2_5/DLFileEntryTypesDDMStructureUpgradeProcess.java#L65-L83)

It will be something like:

```
ObjectFolder objectFolder =
_objectFolderLocalService.getUncategorizedObjectFolder(
						company.getCompanyId());

Role guestRole = _roleLocalService.getRole(
					company.getCompanyId(), RoleConstants.GUEST);
```